### ISSUE TITLE
Notifications.getExpoPushTokenAsync() never resolve

### ISSUE THREAD

**Issue by ozerty** [expo/expo#7606]
## 🐛 Bug Report

The promise `await Notifications.getExpoPushTokenAsync()` never resolve, and there is no error.
It was working the previous week.

### Environment

  Device: Android 9 / OnePlus5

  Expo CLI 3.17.11 environment info:
    System:
      OS: Windows 10 10.0.18363
    Binaries:
      Node: 10.16.3 - C:\Program Files\nodejs\node.EXE
      Yarn: 1.22.0 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
      npm: 6.14.2 - C:\Program Files\nodejs\npm.CMD
    IDEs:
      Android Studio: Version  3.6.0.0 AI-192.7142.36.36.6241897
    npmPackages:
      expo: ~36.0.2 => 36.0.2
      react: 16.9.0 => 16.9.0
      react-native: https://github.com/expo/react-native/archive/sdk-36.0.1.tar.gz => 0.61.4
      react-navigation: ^4.0.10 => 4.1.1

**Comment by Rohansi** [comment]
Getting this in my SDK 35 app after updating to client v2.15.0. Looks to be Android only as my friends with iOS and the same client version have no issues.

Haven't tried standalone SDK 37 yet but will soon.

**Comment by Triggybet** [comment]
I can also reproduce this probblem with SDK35 and the latest Expo clients.

**Comment by cruzach** [comment]
Can you please provide a repro? I tested this on the most recent android client with this snack, and `getExpoPushTokenAsync` is working correctly - https://snack.expo.io/@charliecruzan/pushnotifications36---?platform=ios

**Comment by Bashin** [comment]
Having the same problem in managed workflow SDK37 on IOS.

**Comment by ozerty** [comment]
Hi @cruzach I will need some time to create an expo snack. 
I have tried yours on my device and it is not working for me. That is very weird !

**Comment by cruzach** [comment]
You may want to double-check the SDK version you're using?

**Comment by Bashin** [comment]
I am using SDK 37 bare workflow, IOS only.

**Comment by Bashin** [comment]
#7617 same issue

**Comment by Rohansi** [comment]
@cruzach I just tried that snack on my device and it does reproduce the problem for me. Used it as is, with Android client 2.15.0, and it was unable to receive notifications.

To show that `Notifications.getExpoPushTokenAsync` is never resolving I then added `console.log('getting token')` before that function call and ran it again. That message is printed into the logs but the token is not.

**Comment by christianjuth** [comment]
Possibly related. I have a TestFlight app crashing when I turn on notifications in my app (which calls `Notifications.getExpoPushTokenAsyn`). The app works fine in the Expo CLI with notifications, but the standalone version crashes. Unfortunately, this makes it more difficult to debug since its a production build. Sentry is not catching the error, but I will try adding a try/catch statement. The app was working fine on SDK 36, but as soon as I updated to SDK 37 (with very few other changes to the app) this started happening. I will continue to debug

**Comment by Meinl** [comment]
Yes, same here. I started to developing my app before than last expo android app update (2.15.0) and working fine. Later, I updated on android, and my code kept working as expected, but yesterday, I deleted all data stored on expo app (caché, etc...) and now stucks on retrieving ExpoToken. So, @cruzach, you could trying to delete all data from expo app and try again. I started a new blank project (SDK 37) and doesn't work too.

**Comment by merikwesterveld** [comment]
Having the same issue. Configure Firebase and all but still nog resolving or rejecting the method. Not running on an emulator either using Expo 37 and Expo client 2.15.0 on Anrdoid.

**Comment by sjchmiela** [comment]
### Short story

Hey all, I think I have found the solution to this problem. We will try to resolve the problem ASAP.

### Long story

We use Firebase as the notifications backend, which publishes its own native libraries that help developers use the service. For SDK37 we have upgraded those dependencies (regular thing), but in the throes of the release we haven't taken the time to read the CHANGELOGs from cover to cover.

The `com.google.firebase:firebase-messaging:20.1.2` with which SDK37 has shipped (so Expo client 2.15.0 too) has introduced a change, which we didn't account for — it uses another native library, Firebase Installations SDK to identify devices. In most scenarios this isn't a problem, but in our case, in which the API key is restricted to only the services we know should be allowed, [it may break Firebase Cloud Messaging](https://firebase.google.com/support/release-notes/android#2020-02-27), which, unfortunately, it did.

To mitigate this we will either downgrade the library to `20.1.0` (I have confirmed it fixes the issue) and release an updated version of Expo client or update the project's configuration and hope it fixes the problem over the air.

Resources:
- https://stackoverflow.com/a/58496014/1123156
- https://stackoverflow.com/a/60565682/1123156
- https://stackoverflow.com/a/60698697/1123156
- https://firebase.google.com/support/release-notes/android#2020-02-27

**Comment by sjchmiela** [comment]
As a temporary workaround you could create/go to your Firebase project, configure Cloud Messaging service, download `google-services.json` and add it to your Expo project following the guide at [Setup Native Firebase](https://docs.expo.io/versions/latest/guides/setup-native-firebase/#android). 🙌 

It seems that then the Firebase Installations SDK registers the device under your Firebase project (not the not-yet-ready Expo-provided one), which lets Firebase issue a device push token, which then may be translated to Expo push token, with which `getExpoPushTokenAsync` will resolve.

If it still doesn't work and the API key that the app is using has some restrictions applied, you may need to add "Firebase Installations SDK" to allowed services, as described in [this SO answer](https://stackoverflow.com/a/58496014/1123156).

**Comment by ozerty** [comment]
Hi @sjchmiela Good to know that you identitfy the issue
I will wait the new version of the expo client to try it.
Thanks for your time all.

**Comment by casoftcol** [comment]
Thanks @sjchmiela 
Im waiting the update and review to you.

if you need testers, im can help to you

**Comment by Meinl** [comment]
> ### Short story
> Hey all, I think I have found the solution to this problem. We will try to resolve the problem ASAP.
> 
> ### Long story
> We use Firebase as the notifications backend, which publishes its own native libraries that help developers use the service. For SDK37 we have upgraded those dependencies (regular thing), but in the throes of the release we haven't taken the time to read the CHANGELOGs from cover to cover.
> 
> The `com.google.firebase:firebase-messaging:20.1.2` with which SDK37 has shipped (so Expo client 2.15.0 too) has introduced a change, which we didn't account for — it uses another native library, Firebase Installations SDK to identify devices. In most scenarios this isn't a problem, but in our case, in which the API key is restricted to only the services we know should be allowed, [it may break Firebase Cloud Messaging](https://firebase.google.com/support/release-notes/android#2020-02-27), which, unfortunately, it did.
> 
> To mitigate this we will either downgrade the library to `20.1.0` (I have confirmed it fixes the issue) and release an updated version of Expo client or update the project's configuration and hope it fixes the problem over the air.
> 
> Resources:
> 
> * https://stackoverflow.com/a/58496014/1123156
> * https://stackoverflow.com/a/60565682/1123156
> * https://stackoverflow.com/a/60698697/1123156
> * https://firebase.google.com/support/release-notes/android#2020-02-27

This is just for FCM?, or fix too retrieve ExpoToken from device?. I can't get ExpoToken from device, blank project, using Permissions.askAsync(Permissions.NOTIFICATIONS, never resolve. Thanks in advance.

**Comment by casoftcol** [comment]
> > ### Short story
> > Hey all, I think I have found the solution to this problem. We will try to resolve the problem ASAP.
> > ### Long story
> > We use Firebase as the notifications backend, which publishes its own native libraries that help developers use the service. For SDK37 we have upgraded those dependencies (regular thing), but in the throes of the release we haven't taken the time to read the CHANGELOGs from cover to cover.
> > The `com.google.firebase:firebase-messaging:20.1.2` with which SDK37 has shipped (so Expo client 2.15.0 too) has introduced a change, which we didn't account for — it uses another native library, Firebase Installations SDK to identify devices. In most scenarios this isn't a problem, but in our case, in which the API key is restricted to only the services we know should be allowed, [it may break Firebase Cloud Messaging](https://firebase.google.com/support/release-notes/android#2020-02-27), which, unfortunately, it did.
> > To mitigate this we will either downgrade the library to `20.1.0` (I have confirmed it fixes the issue) and release an updated version of Expo client or update the project's configuration and hope it fixes the problem over the air.
> > Resources:
> > 
> > * https://stackoverflow.com/a/58496014/1123156
> > * https://stackoverflow.com/a/60565682/1123156
> > * https://stackoverflow.com/a/60698697/1123156
> > * https://firebase.google.com/support/release-notes/android#2020-02-27
> 
> This is just for FCM?, or fix too retrieve ExpoToken from device?. I can't get ExpoToken from device, blank project, using Permissions.askAsync(Permissions.NOTIFICATIONS, never resolve. Thanks in advance.

The problem is with ExpoToken!
For this your problem!
